pennylane/tape/tapes/cv_param_shift.py

```diff
@@ -368,27 +378,48 @@ def parameter_shift(self, idx, device, params, **options):
             1st order method may be more efficient unless it's really easy to
             experimentally measure arbitrary 2nd order observables.
 
+        .. warning::
+
+            The 2nd order method can only be executed on devices that support the
+            :class:`~.PolyXP` observable.
+
         Args:
             idx (int): trainable parameter index to differentiate with respect to
-            device (.Device): a PennyLane device that can execute quantum operations and return
-                measurement statistics
             params (list[Any]): the quantum tape operation parameters

```

Comment:
`params` now a keyword argument

Also, I didn't see why that change was necessary for the code below :thinking:

Reply:
Ah, this was a change that was 'forced' by pylint due to https://en.wikipedia.org/wiki/Liskov_substitution_principle.

I'll go through the code and standardize this everywhere to not be a keyword argument.